### PR TITLE
Closes part of #682

### PR DESCRIPTION
**Pull request by shamikbose**
medical_data updated to `bigbio_pairs` schema
Passes all tests

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: bigbio/biodatasets/medical_data/medical_data.py [hunk 1]
```diff
@@ -146,33 +145,35 @@ def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
...
+        elif self.config.schema == "bigbio_pairs":
             csv_reader = pd.read_csv(filepath, dtype="object")
             # Train and test splits handled differently as test split is missing values
             if split == "train":
                 for _cols, line in csv_reader.iterrows():
                     document = {}
                     document["id"] = line["unique_hash"]
-                    document["premise"] = line["text"]
-                    document["hypothesis"] = line["drug"]
+                    document["document_id"] = "NULL"
```
Comment (galtay) [inline bigbio/biodatasets/medical_data/medical_data.py:166]: if there is no document id provided by the dataset lets make it equal to `id`

Comment (shamikbose) [reply bigbio/biodatasets/medical_data/medical_data.py:166]: Done

**Comment by mariosaenger** [comment]
@phlobo Please have a look at this dataset. I refactored the implementation to the new HF-hub-based style. I'm not completely happy with the modeling - essentially it's aspect-based text classification rather than text similarity. However, the former isn't properly supported by any of the BigBio schemes.

File: bigbio/hub/hub_repos/samd/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- English
+license: UNKNOWN
+multilinguality: monolingual
+bigbio_license_shortname: UNKNOWN
+pretty_name: Sentiment Analysis for Medical Drugs (SAMD)
+homepage: https://www.kaggle.com/datasets/arbazkhan971/analyticvidhyadatasetsentiment
+bigbio_pubmed: False
+bigbio_public: False
+bigbio_tasks: 
+- SEMANTIC_SIMILARITY
```
Comment (phlobo) [inline bigbio/hub/hub_repos/samd/README.md:15]: @mariosaenger I think the schema is alright, but maybe the task "TEXT_PAIRS_CLASSIFICATION" is more appropriate (as it is more generic)?

File: bigbio/hub/hub_repos/samd/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+# Dataset Card for Sentiment Analysis for Medical Drugs (SAMD)
+
+## Dataset Description
+
+- **Homepage:** https://www.kaggle.com/datasets/arbazkhan971/analyticvidhyadatasetsentiment
+- **Pubmed:** False
+- **Public:** False
+- **Tasks:** STS
```
Comment (phlobo) [inline bigbio/hub/hub_repos/samd/README.md:26]: same here

File: bigbio/hub/hub_repos/samd/samd.py [hunk 4]
```diff
@@ -36,32 +34,36 @@
 }
 """
 
-_DATASETNAME = "medical_data"
-_DISPLAYNAME = "Medical Data"
+_DATASETNAME = "samd"
+_DISPLAYNAME = "Sentitment Analyis for  Medical Drugs"
```
Comment (phlobo) [inline bigbio/hub/hub_repos/samd/samd.py:38]: Typo

File: bigbio/hub/hub_repos/samd/samd.py [hunk 5]
```diff
@@ -36,32 +34,36 @@
...
 
-_LICENSE = Licenses.UNKNOWN
+_HOMEPAGE = "https://www.kaggle.com/datasets/arbazkhan971/analyticvidhyadatasetsentiment"
+_LICENSE = "UNKNOWN"
 
 _URLS = {}
 
-_SUPPORTED_TASKS = [Tasks.TEXTUAL_ENTAILMENT]
-
+_SUPPORTED_TASKS = [Tasks.SEMANTIC_SIMILARITY]
```
Comment (phlobo) [inline bigbio/hub/hub_repos/samd/samd.py:53]: see comment above

**Review by phlobo — COMMENTED** [review]
I would pick a different task, but apart from that, the PR looks good!

File: bigbio/hub/hub_repos/samd/README.md (hunk 2 above)
Comment (phlobo) [reply bigbio/hub/hub_repos/samd/README.md:15]: See also: https://github.com/bigscience-workshop/biomedical/issues/682#issuecomment-1146717432

Comment (mariosaenger) [reply bigbio/hub/hub_repos/samd/README.md]: Thanks for the hint. Revised it to TEXT_PAIRS_CLASSIFICATION.

File: bigbio/hub/hub_repos/samd/README.md (hunk 3 above)
Comment (mariosaenger) [reply bigbio/hub/hub_repos/samd/README.md]: Fixed.

File: bigbio/hub/hub_repos/samd/samd.py (hunk 5 above)
Comment (mariosaenger) [reply bigbio/hub/hub_repos/samd/samd.py]: Fixed.

**Review by phlobo — DISMISSED** [review]

**Review by phlobo — APPROVED** [review]

**Comment by phlobo** [comment]
@mariosaenger I realized the old version of the dataset is already on the hub: https://huggingface.co/datasets/bigbio/medical_data
should we delete that one?

**Pull request merged**